NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1032 @@
+            electric_field.set_trace_start_time(trace_start_time)
+            electric_field[efp.azimuth] = azimuth
+            electric_field[efp.zenith] = zenith
+            electric_field[efp.ray_path_type] = propagation.solution_types[p.get_solution_type(iS)]
+            electric_field[efp.nu_vertex_distance] = R
+            electric_field[efp.nu_vertex_travel_time] = T
+            electric_field[efp.raytracing_solution] = p.get_raytracing_output(iS)
+            electric_field[efp.launch_vector] = p.get_launch_vector(iS)
+
+            if min_efield_amplitude is not None:
+                if np.max(np.abs(electric_field.get_trace())) > min_efield_amplitude:
+                    sim_station.set_candidate(True)
+
+            sim_station.add_electric_field(electric_field)
+    return sim_station
+
+def apply_det_response_sim(sim_station, det, config,
+                        detector_simulation_filter_amp=None,
+                        evt=None,
+                        event_time=None,
+                        detector_simulation_part1=None,
+                        time_logger=None):
+    """
+    Apply the detector response to the simulated electric field, i.e., the voltage traces
+    seen by the readout system, per shower, raytracing solution and channel.
+    This includes the effect of the antenna response, the
+    analog signal chain. The result is a list of SimChannel objects which are added to the
+    SimStation object.
+
+
+    Parameters
+    ----------
+    sim_station : sim_station object that contains the electric fields at the observer positions
+        A list of SimEfield objects, one for each shower and propagation solution
+    det : Detector object
+        the detector description defining all channels.
+    config : dict
+        the NuRadioMC configuration dictionary (from the yaml file)
+    detector_simulation_filter_amp: function (optional)
+        a function that applies the filter and amplifier response to the electric field
+        the arguments to the function are (event, station, detector)
+        if not provided, the function `detector_simulation_part1` needs to be provided.
+    evt : NuRadioReco event object (optional)
+        all NuRadioReco modules that get executed will be registered to the event.
+        If no event is provided, a dummy event is created so that the function runs, but
+        then this information is not available to the user. 
+    event_time: time object (optional)
+        the time of the event to be simulated
+    detector_simulation_part1: function (optional)
+        this function gives the user the full flexibility to implement all processing
+        arguments to the function are (event, station, detector)
+
+    Returns nothing. The SimChannels are added to the SimStation object.
+    """
+    if time_logger is not None: time_logger.start_time('detector response (sim)')
+    if evt is None:
+        evt = NuRadioReco.framework.event.Event(0, 0)
```

Comment:
`evt = evt or NuRadioReco.framework.event.Event(0, 0)`

Reply:
I'm not a big fan of this - you save one line of code at the cost of making it less straightforward to parse for a human.

Reply:
maybe I got so used to this syntax but for me it makes it easier to read code ...